Xamarin.Essentials/Platform/Platform.ios.tvos.watchos.cs

```diff
@@ -138,5 +146,21 @@ internal static UIWindow GetCurrentWindow(bool throwIfNull = true)
 
         internal static NSOperationQueue GetCurrentQueue() =>
             NSOperationQueue.CurrentQueue ?? new NSOperationQueue();
+
+#if __IOS__
+        internal class UIPresentationControllerDelegate : UIAdaptivePresentationControllerDelegate
+        {
+            public Action DismissHandler { get; set; }
+
+            public override void DidDismiss(UIPresentationController presentationController) =>
+                DismissHandler?.Invoke();
```

Comment:
Can we set this to null ofter invoking? This way we don't get a double finish. Same with the Dispose invocation. I think we are fine now because all our usage is with a TrySetResult, so it makes no difference at the moment. But just in case there are other things in the future it is good to be safe.